src/cne/elector.rs

```diff
@@ -17,6 +39,316 @@ pub struct Elector {
 
 impl From<String> for Elector {
   fn from(html_string: String) -> Self {
-    todo!()
+
+    if verify_citizenship_error(&html_string) {
+      panic!("Invalid citizenship provided");
+    }
+
+    let document = Html::parse_fragment(&html_string);
+    let selector = Selector::parse("table tbody tr td table tbody tr:last-child td table tbody tr:nth-child(2) td table:first-child tbody tr").unwrap();
+    let mut selection = document.select(&selector);
+
+    let citizenship = select_field(&mut selection).inner_html();
+
+    Elector {
+      citizenship: match citizenship.chars().next().unwrap() {
+        'E' => Citizenship::E,
+        _ => Citizenship::V,
+      },
+      identity: citizenship
+        .chars()
+        .skip(2)
+        .take(citizenship.len() - 2)
+        .collect(),
+      name: select_bold_text(&mut select_field(&mut selection)).inner_html(),
+      state: select_field(&mut selection).inner_html(),
+      town: select_field(&mut selection).inner_html(),
+      parish: select_field(&mut selection).inner_html(),
+      voting_center: select_font_tag(&mut select_field(&mut selection)).inner_html(),
+      address: select_font_tag(&mut select_field(&mut selection)).inner_html(),
+    }
+  }
+}
+
+#[cfg(test)]
```

Comment:
Test are now set on the `master` branch, we could move this to the tests directory